io.github.MCDFSteve.NipaPlay-Reload.yaml

```
@@ -0,0 +1,129 @@
+  # Audio
+  - --socket=pulseaudio
+  # Network access
+  - --share=network
+  # D-Bus access (for portals, etc.)
+  - --talk-name=org.freedesktop.portal.Desktop
+  - --talk-name=org.freedesktop.portal.Documents
+  - --talk-name=org.freedesktop.portal.FileChooser
+
+# Placeholder for the commit or tag for your application source
+# Flathub will typically build from a tag like 'v0.3.10'
+# You'll replace this placeholder or Flathub's build system will manage it.
+# x-checker-data:
+#   type: git
+#   url: https://github.com/MCDFSteve/NipaPlay-Reload.git
+#   tag-pattern: ^v([\d.]+)$ # Example tag pattern
+
+modules:
+  # Module to download and setup Flutter SDK
+  - name: flutter-sdk
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/flutter-sdk # Changed from /app/flutter to avoid potential name clashes if app itself is named flutter
+      - curl -L -o flutter.tar.xz "https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_3.29.3-stable.tar.xz"
+      - tar -xf flutter.tar.xz -C /app/flutter-sdk --strip-components=1 # Extract directly into flutter-sdk and strip leading 'flutter' directory
+      # No install commands needed as it's used by the next module directly from /app/flutter-sdk
+    sources: [] # No external sources, done via curl
+
+  # Application Module
+  - name: nipaplay-reload
+    buildsystem: simple
+    # Sources: Get source code from your GitHub repository
+    # For local testing, you might use:
+    # sources:
+    #  - type: dir
+    #    path: . # If manifest is in the root of your project
+    # For Flathub, it will typically be git:
+    sources:
+      - type: git
+        url: https://github.com/MCDFSteve/NipaPlay-Reload.git
+        # This commit/tag will be overridden by Flathub's build system
+        # based on the release tag (e.g., v0.3.10)
+        # For local testing, you can specify a commit or branch:
+        # commit: main
+        # Or for a specific tag for testing:
+        tag: v0.3.10 # <<<< IMPORTANT: REPLACE WITH THE ACTUAL TAG/COMMIT FOR THE RELEASE
+    build-commands:
+      # Setup Flutter SDK path from the manually downloaded SDK
+      - export PATH="/app/flutter-sdk/bin:$PATH"
+      # Verify Flutter is found
+      - flutter --version
+      # Standard Flutter build for Linux
+      - flutter pub get
+      - flutter build linux --release
+      # Install the application files
+      - mkdir -p /app/bin/
+      - mkdir -p /app/lib/
+      - mkdir -p /app/share/nipaplay/
+      # Copy the main executable
+      - cp build/linux/x64/release/bundle/NipaPlay /app/bin/nipaplay # Ensure target name is 'nipaplay'
+      # Copy data and lib folders
+      - cp -r build/linux/x64/release/bundle/data /app/share/nipaplay/
+      - cp -r build/linux/x64/release/bundle/lib /app/lib/
+      
+      # Install .desktop file
+      - mkdir -p /app/share/applications/
+      - cp assets/linux/io.github.MCDFSteve.NipaPlay-Reload.desktop /app/share/applications/
```

```suggestion
      - install -Dm644 assets/linux/io.github.MCDFSteve.NipaPlay-Reload.desktop  -t /app/share/applications/
```

Use `install`. This also apply further down.